src/testdir/test_listdict.vim

```diff
@@ -1619,4 +1619,28 @@ func Test_deep_nested_listdict_compare()
   call v9.CheckLegacyAndVim9Success(lines)
 endfunc
 
+" Test for using id()
+def Test_id_with_dict()
+  # demonstate a way that "id(item)" differs from "string(item)"
+  var lines =<< trim END
+    vim9script
+    var d1 = {one: 1}
+    var d2 = {one: 1}
+    var d3 = {one: 1}
+    var idDict: dict<any>
+    idDict[id(d1)] = d1
+    idDict[id(d2)] = d2
+    idDict[id(d3)] = d3
+    assert_equal(3, idDict->len())
+
+    var stringDict: dict<any>
+    stringDict[string(d1)] = d1
+    stringDict[string(d2)] = d2
+    stringDict[string(d3)] = d3
+    assert_equal(1, stringDict->len())
+
+    assert_equal('', id(3))
+  END
+  v9.CheckSourceSuccess(lines)
```

Comment:
Using `CheckSourceSuccess` here is a bit strange here, as for this function it's not really different from putting these line in the test function directly, and unlike `CheckLegacyAndVim9Success` it only runs these lines once.

Reply:
I'm fuzzy about the test infrastructure. I guess I was thinking that using `v9.Check*` also gives you a context so you don't need to clean up. I can make that change. I guess `:unlet ...` is required then?

Reply:
Oh, in any event, it should have been `CheckScriptSuccess()` (copy/paste blunder).

Reply:
> Oh, in any event, it should have been `CheckScriptSuccess()` (copy/paste blunder).

Ah, but wait... I just looked a little deeper into `v9.CheckSource` vs `v9.CheckScript`. I'm now thinking that `v9.CheckSource` is correct here; it starts with a new window, puts the code in it, sources the code from the buffer; `:bwipes` the buffer when done; whereas `v9CheckScript` creates a file.

Reply:
> I'm fuzzy about the test infrastructure. I guess I was thinking that using `v9.Check*` also gives you a context so you don't need to clean up. I can make that change. I guess `:unlet ...` is required then?

The variables here are all local variables, so no `:unlet` is required if you put the lines directly in the function.